BraveRewardsTests/LedgerDatabaseTests.mm

```diff
@@ -144,6 +147,638 @@ - (void)testNonZeroExcludedCount
+  }];
+}
+
+#pragma mark - Activity Info
+
+- (void)testInsertOrUpdateActivityInfoFromPublisherEmptyId
+{
+  const auto info = [self createBATPublisherInfo:@"111" reconcileStamp:111 percent:11 createActivityInfo:NO];
+  XCTAssert([self countMustBeEqualTo:0 forEntityName:@"ActivityInfo"]);
+  
+  info.id = @"";
+  [self waitForFailedSaveAttempt:^{
+    [BATLedgerDatabase insertOrUpdateActivityInfoFromPublisher:info];
+  }];
+  XCTAssert([self countMustBeEqualTo:0 forEntityName:@"ActivityInfo"]);
+}
+
+- (void)testInsertOrUpdateActivitiesInfoFromPublishers
+{
+  const auto p1 = [self createBATPublisherInfo:@"111" reconcileStamp:111 percent:11 createActivityInfo:NO];
+  const auto p2 = [self createBATPublisherInfo:@"222" reconcileStamp:222 percent:22 createActivityInfo:NO];
+  const auto p3 = [self createBATPublisherInfo:@"333" reconcileStamp:333 percent:33 createActivityInfo:NO];
+  
+  const auto publishers = @[p1, p2, p3];
+  
+  XCTAssert([self countMustBeEqualTo:0 forEntityName:@"ActivityInfo"]);
+  
+  [self backgroundSaveAndWaitForExpectation:^{
+    [BATLedgerDatabase insertOrUpdateActivitiesInfoFromPublishers:publishers];
+  }];
+  
+  XCTAssert([self countMustBeEqualTo:3 forEntityName:@"ActivityInfo"]);
+}
+
+- (void)testDeleteActivityInfoWithPublisherID
+{
+  [self createBATPublisherInfo:@"1" reconcileStamp:10 percent:11 createActivityInfo:YES];
+  [self createBATPublisherInfo:@"2" reconcileStamp:10
+                                               percent:22 createActivityInfo:YES];
+  
+  XCTAssert([self countMustBeEqualTo:2 forEntityName:@"ActivityInfo"]);
+  
+  [self backgroundSaveAndWaitForExpectation:^{
+    [BATLedgerDatabase deleteActivityInfoWithPublisherID:@"1" reconcileStamp:10];
+  }];
+  XCTAssert([self countMustBeEqualTo:1 forEntityName:@"ActivityInfo"]);
+}
+
+#pragma mark - publishersWithActivityFromOffset
+
+- (void)testPublishersWithFiltersEmptyFilter
+{
+  const auto offset = 0;
+  const auto limit = 0;
+  const auto filter = [[BATActivityInfoFilter alloc] init];
+  
+  [self createBATPublisherInfo:@"1" reconcileStamp:10 percent:10 createActivityInfo:true];
+  [self createBATPublisherInfo:@"2" reconcileStamp:10 percent:10 createActivityInfo:true];
+  
+  const auto result = [BATLedgerDatabase publishersWithActivityFromOffset:offset limit:limit filter:filter];
+  
+  XCTAssertEqual(result.count, 2);
+}
+
+- (void)testPublishersWithFiltersLimitOffset
+{
+  const auto filter = [[BATActivityInfoFilter alloc] init];
+  
+  [self createBATPublisherInfo:@"1" reconcileStamp:10 percent:10 createActivityInfo:true];
+  [self createBATPublisherInfo:@"2" reconcileStamp:10 percent:10 createActivityInfo:true];
+  [self createBATPublisherInfo:@"3" reconcileStamp:10 percent:10 createActivityInfo:true];
+  
+  const auto limit = 1;
+  const auto resultWithLimit = [BATLedgerDatabase publishersWithActivityFromOffset:0 limit:limit filter:filter];
+  
+  XCTAssertEqual(resultWithLimit.count, 1);
```

Comment:
Same comment as above in regards to testing only count instead of result of the action (in this case, querying an item).  Maybe intended based on the test name?